src/WindowsTracing/ContextSwitchManagerTest.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "ContextSwitchManager.h"
+#include "MockTracerListener.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+
+namespace orbit_windows_tracing {
+
+TEST(ContextSwitch, ListenerIsCalled) {
+  orbit_windows_tracing::MockTracerListener mock_listener;
+  ContextSwitchManager manager(&mock_listener);
+
+  EXPECT_CALL(mock_listener, OnSchedulingSlice).Times(testing::Exactly(1));
+
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/1, /*new_tid=*/2, /*timestamp_ns=*/0);
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/2, /*new_tid=*/1, /*timestamp_ns=*/1);
+}
+
+TEST(ContextSwitch, InvalidPidIsSet) {
+  orbit_windows_tracing::MockTracerListener mock_listener;
+  orbit_windows_tracing::FakeTracerListener& fake_listener = mock_listener.fake_;
+  mock_listener.DelegateToFake();
+  ContextSwitchManager manager(&mock_listener);
+
+  EXPECT_CALL(mock_listener, OnSchedulingSlice).Times(testing::Exactly(1));
+
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/1, /*new_tid=*/2, /*timestamp_ns=*/0);
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/2, /*new_tid=*/1, /*timestamp_ns=*/1);
+
+  EXPECT_EQ(fake_listener.scheduling_slices_.size(), 1);
+  EXPECT_EQ(fake_listener.scheduling_slices_[0].pid(), orbit_base::kInvalidProcessId);
+  EXPECT_EQ(fake_listener.scheduling_slices_[0].tid(), 2);
+}
+
+TEST(ContextSwitch, ValidPidIsSet) {
+  orbit_windows_tracing::MockTracerListener mock_listener;
+  orbit_windows_tracing::FakeTracerListener& fake_listener = mock_listener.fake_;
+  mock_listener.DelegateToFake();
+  ContextSwitchManager manager(&mock_listener);
+
+  EXPECT_CALL(mock_listener, OnSchedulingSlice).Times(testing::Exactly(1));
+
+  manager.ProcessThreadEvent(/*tid*/ 2, /*pid=*/3);
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/1, /*new_tid=*/2, /*timestamp_ns=*/0);
+  manager.ProcessCpuEvent(/*cpu=*/0, /*old_tid=*/2, /*new_tid=*/1, /*timestamp_ns=*/1);
+
+  EXPECT_EQ(fake_listener.scheduling_slices_.size(), 1);
+  EXPECT_EQ(fake_listener.scheduling_slices_[0].pid(), 3);
+}
+
```

Comment:
I'd probably add another test, that checks for multiple scheduling slices.

Reply:
Done